test/E2ETest/testassets/projects/multi-targetframeworks-and-targetframework-too/multi-targetframeworks-and-targetframework-too/multi-targetframeworks-and-targetframework-too.csproj

```diff
@@ -0,0 +1,13 @@
+<Project Sdk="Microsoft.NET.Sdk.Web">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
```

Comment:
projects shouldn't specify both `TargetFramework` and `TargetFrameworks` at the same time. I don't think testing such a scenario is needed.

Reply:
As I understand it's weird to read, this is how `MsBuild` and `dotnet` works today
So I tried to cover existing "possible scenario"
This is also how `dotnet/tye` works in `tye.yaml` when using `buildProperties` for `TargetFramework` if the csproj uses `TargetFrameworks`
What should I do ? Do you want me to remove this sample anyway ?

eg:
![image](https://user-images.githubusercontent.com/2266487/91977976-b1274300-ed23-11ea-8a65-841417178291.png)

Reply:
So yes and no, for example take a look at the docs https://docs.microsoft.com/en-us/dotnet/standard/frameworks#how-to-specify-target-frameworks. The idea is that you can target multiple frameworks in your project by using `TargetFrameworks`. This is especially important for libraries which might need to maintain compatibility with older frameworks but also take advantage of new APIs but can also be done for apps like you've demonstrated. However, if you do multi-target apps you are essentially building for both TFMs and you will need to specify a specific target framework to run. The mechanism to do this is to specify a single `TargetFramework`, for example via CLI. However, it doesn't make sense to do so in a csproj file since if you are specifying a single `TargetFramework` at build time, why even bother defining multiple frameworks in the first place?

Reply:
sorry, as I tried to review your comments here, I know realize that my answer is split in two with 
https://github.com/dotnet/tye/pull/567#discussion_r482027153

Also this is also practical aside library, especially when you are migrating from `netcoreapp3.1` to `net5.0` for example it allows you to have both code build / test / E2E and decide which artifact you will ship to production (or even both for canary or blue/green)

Of course the `--framework` is, defacto, mandatory for such use, but it's not limited to library. I did not tried to overthink it to be fair, I just tried to make sure the behavior of `tye` was somehow the same as `dotnet`.
I'm fine with keeping or removing it, but I guess this will depends on the other comment bellow

Reply:
As I said previously,
> but can also be done for apps like you've demonstrated
I'm agreeing with you that there is valid use of multi-targeted apps. The comment about libraries is more of a side note. My main point is that it doesn't make sense to specify both TargetFramework and TargetFrameworks in an app. You either specify one or the other.

Reply:
Just to circle back, the `TargetFramework` property should be removed here.